ec2_terminator/app.py

```diff
@@ -0,0 +1,157 @@
+    state_filters = [{
+        'Name': 'instance-state-name',
+        'Values': ['running', 'stopping', 'stopped'],
+    }]
+
+    instances = []
+
+    pager = client.get_paginator('describe_instances')
+    for page in pager.paginate(Filters=state_filters):
+        for rsvp in page['Reservations']:
+            for ec2 in rsvp['Instances']:
+                ec2_id = ec2['InstanceId']
+                logging.debug(f"EC2: {ec2}")
+                instances.append(ec2_id)
+
+    logging.debug(f"Instances found: {instances}")
+    return instances
+
+
+def stop_instances(instances, region):
+    """Stop all given instances"""
+    client = boto3.client(region_name=region)
+
+    stopped = []
+    resp = client.stop_instances(InstanceIds=instances)
+    if resp['StoppingInstances']:
+        stopped = [s['InstanceId'] for s in resp['StoppingInstances']]
+
+    logging.debug(f"Stopped: {stopped}")
+    return stopped
+
+
+def terminate_instances(instances, region):
+    """Terminate all given instances"""
+    client = boto3.client(region_name=region)
+
+    terminated = []
+    resp = client.terminate_instances(InstanceIds=instances)
+    if resp['TerminatingInstances']:
+        terminated = [t['InstanceId'] for t in resp['TerminatingInstances']]
+
+    logging.debug(f"Terminated :{terminated}")
+    return terminated
+
+
+def lambda_handler(event, context):
+    """Lambda function to stop or terminate all EC2 instances in the current  account.
+
+    Parameters
+    ----------
+    event: dict, required
+        API Gateway Lambda Proxy Input Format
+
+        Event doc: https://docs.aws.amazon.com/apigateway/latest/developerguide/set-up-lambda-proxy-integrations.html#api-gateway-simple-proxy-for-lambda-input-format
+
+    context: object, required
+        Lambda Context runtime methods and attributes
+
+        Context doc: https://docs.aws.amazon.com/lambda/latest/dg/python-context-object.html
+
+    Returns
+    ------
+    API Gateway Lambda Proxy Output Format: dict
+
+        Return doc: https://docs.aws.amazon.com/apigateway/latest/developerguide/set-up-lambda-proxy-integrations.html
+    """
+
+    try:
+        # Are we stopping or terminating instances?
+        ec2_action = 'stop'
+        if os.environ.get('EC2_ACTION', '') == 'TERMINATE':
+            ec2_action = 'terminate'
+
+        # Get the list of regions
+        regions = list_regions()
+        if not regions:
+            raise ValueError("No available regions")
+
+        # List of stopped or terminated instances
+        found = False
+        processed = []
+
+        # Iterate over every region
+        for region in regions:
+            logging.info(f"Region: {region}")
+            ec2_instances = list_instances(region)
+
+            # Stop or terminate any instances found
+            if ec2_instances:
+                found = True
+                if ec2_action == 'terminate':
+                    logging.info(f"Terminating Instances: {ec2_instances}")
```

Comment:
should this be at debug level since there's already logging at info level below with `logging.info(message)`?

Reply:
done